src/python_testing/TC_EGCTestBase.py

```diff
@@ -0,0 +1,81 @@
+#
+#    Copyright (c) 2025 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+
+import logging
+from datetime import datetime, timedelta, timezone
+
+import chip.clusters as Clusters
+from chip.clusters.Types import NullValue
+from chip.testing import matter_asserts
+from mobly import asserts
+
+logger = logging.getLogger(__name__)
+
+
+class ElectricalGridConditionsTestBaseHelper:
+
+    # Spec derived constants
+    kMaxForecastEntries = 56    # Maximum number of list entries for Forecasts
+
+    # Test event trigger IDs
+    kEventTriggerCurrentConditionsUpdate = 0x00A0000000000000
+    kEventTriggerForecastConditionsUpdate = 0x00A0000000000001
+
+    def check_ForecastConditions(self,
+                                 cluster: Clusters.ElectricalGridConditions = Clusters.ElectricalGridConditions,
+                                 forecastConditions: list = None):
```

Comment:
```suggestion
    def check_ForecastConditions(self,
                                 cluster: Clusters.ElectricalGridConditions,
                                 forecastConditions: list):
```

We seem to always call this with parameters set, so no need to provide defaults (otherwise type is wrong: None would not be a list but rather `Optional[list]`.

Reply:
Done

Reply:
Done